Automattic/jetpack
# Publicize: always add an alt text to Open Graph Images

Fixes #18307 

#### Changes proposed in this Pull Request:

* Jetpack now always outputs an `og:image:alt` tag when an `og:image` tag is present.
* Allows output of empty alt tag.
* Uses the alt text of an image if possible (attachment page or page/post with extracted image). On author pages with an avatar, the display name of the author is used.
* In all other cases, falls back to the default.
* Default can be changed using the `jetpack_open_graph_image_default_alt_text` filter.

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Publish a page or post and add an image.
* Look at the source code and check if `og:image` and `og:image:alt` are present in the Jetpack Open Graph tags.
* Try changing the alt text of the image in the post and ensure that it's reflected in the `og:image:alt` tag.
* Remove the alt text of the image and verify an empty `og:image:alt` tag is still present.
* Use the `jetpack_open_graph_image_default_alt_text` filter to add a new default text, like so:
```php
add_filter( 'jetpack_open_graph_image_default_alt_text', function() {
	return 'Default alt text';
} );
```
